core/chaincode/exectransaction.go

```diff
@@ -139,6 +140,9 @@ func ExecuteTransactions(ctxt context.Context, cname ChainName, xacts []*pb.Tran
 	txerrs = make([]error, len(xacts))
 	for i, t := range xacts {
 		_, txerrs[i] = Execute(ctxt, chain, t)
+		if txerrs[i] != nil {
+			sendTxRejectedEvent(xacts[i].Uuid, txerrs[i].Error())
```

Comment:
I don't think this is the right place to emit the event.  You should only emit the event when the errored transaction is (not) being committed to a block.  Until the block is committed, anything is possible.